skill_randomness/__init__.py

```diff
@@ -71,37 +91,36 @@ def handle_fortune_teller(self, message: Message):  # pylint: disable=unused-arg
         self.enclosure.eyes_spin()
         self.enclosure.mouth_text(fortune_with_answer)
 
-    @intent_handler("roll-dice.intent")
-    def handle_roll_dice(self, message: Message):
+    @intent_handler("roll-single-die.intent")
+    def handle_roll_single_die(self, message: Message):
         """Roll a die."""
-        self.log.debug(f"Roll dice message: {message.serialize()}")
+        # self.log.debug(f"Message: {message.serialize()}")
+        faces = extract_number(message.data.get("faces", "6"))
         self.play_audio(f"{dirname(__file__)}/die-roll.wav")
-        number = message.data.get("number", 1)
-        faces = message.data.get("faces", 6)
-        result = 0
-        if int(number) > 1:
-            self.log.debug(f"rolling {number} dice")
-            result_string = ""
-            for _ in range(1, int(number) + 1):
-                # Create a dialog string
-                r = Die(d(int(faces))).sample()
-                result += r
-                result_string = result_string + str(r) + ", "
-            result = result_string + f" for a total of {result}"
-        else:
-            result = Die(d(int(faces))).sample()
+        self.log.debug(f"Rolling a die with {faces} faces")
+        result = Die(d(int(faces))).sample()
         self.speak_dialog("die-result", data={"result": result})
         self.gui.show_text(str(result))
         self.enclosure.eyes_spin()
         self.enclosure.mouth_text(str(result))
 
-    @intent_handler("roll-common-dice.intent")
-    def handle_roll_common_dice(self, message: Message):
-        """Roll traditional six-sided dice."""
+    @intent_handler("roll-multiple-dice.intent")
+    def handle_roll_multiple_dice(self, message: Message):
+        die_limit = self.settings.get("die_limit", 6)
```

Comment:
Please move to a property to read the latest value in the settings:

```py
    @property
    def die_limit(self):
        return self.settings.get("die_limit", 50)
```

I'd suggest a higher default limit than 6 because anyone using this for tabletop RPGs is likely to use at least 16 on a fairly regular basis (D&D 5th edition critical hit on a basic fireball will be 16d6)

Reply:
wow, ok.  No idea it used that many

Reply:
It's kind of fun to watch that many dice fly around 🙂 Shadowrun is another game that's almost entirely run with d6s. It's not uncommon to roll 20+ dice regularly